src/lib/helperFunctions.ts

```diff
@@ -45,3 +45,25 @@ export const statusToNumber = (
   }
   return 0;
 };
+
+export const isHostnameLocal = (hostname: string) => {
+  if (hostname === "localhost" || hostname === "[::1]") {
+    return true;
+  }
+  // Test 127.0.0.0/8 range.
+  const blocks = hostname.split(".");
+  if (blocks.length === 4 && blocks[0] === "127") {
+    return blocks
+      .slice(1)
+      .every((block) => Number(block) >= 0 && Number(block) < 256);
+  }
+  return false;
+};
+
+export const isUriLocalhost = (uri: string) => {
```

Comment:
Careful on return types here, as you've effectively got: `isUriLocalhost(uri: string): boolean | undefined` which would be what's considered a "tri-state boolean", which is totally a misnomer. if you encounter an error, it's probably safe to return `false`

Reply:
changed with https://github.com/inrupt/client-identifier-helper/pull/133/commits/57f3170d9aded65e9e2ac540f765b91b0bdd5be8

Reply:
https://en.wikipedia.org/wiki/Boolean_data_type — _In [computer science](https://en.wikipedia.org/wiki/Computer_science), the Boolean (sometimes shortened to Bool) is a [data type](https://en.wikipedia.org/wiki/Data_type) that has one of two possible values (usually denoted true and false)_

So a value which is true / false / undefined really isn't a boolean, it's some third thing that doesn't generally make sense.